doc/sphinxext/related_software.py

```diff
@@ -0,0 +1,226 @@
+    "fsleyes": {
+        "Home-page": "https://fsl.fmrib.ox.ac.uk/fsl/fslwiki/FSLeyes",
+        "Summary": "FSLeyes is the FSL image viewer.",
+    },
+    # dcm2niix must be built from source
+    "dcm2niix": {
+        "Home-page": "https://github.com/rordenlab/dcm2niix",
+        "Summary": "DICOM to NIfTI converter",
+    },
+    # TODO: mnelab forces PySide6, it can be added to doc_related when we use PySide6
+    # for doc building. Also its package does not set the Home-page property.
+    "mnelab": {
+        "Home-page": "https://github.com/cbrnr/mnelab",
+        "Summary": "A graphical user interface for MNE",
+    },
+    # TODO: these do not set a valid homepage or documentation page on PyPI
+    "mne-features": {
+        "Home-page": "https://mne.tools/mne-features",
+        "Summary": "MNE-Features software for extracting features from multivariate time series",  # noqa: E501
+    },
+    "mne-rsa": {
+        "Home-page": "https://users.aalto.fi/~vanvlm1/mne-rsa",
+        "Summary": "Code for performing Representational Similarity Analysis on MNE-Python data structures.",  # noqa: E501
+    },
+    "mffpy": {
+        "Home-page": "https://github.com/BEL-Public/mffpy",
+        "Summary": "Reader and Writer for Philips' MFF file format.",
+    },
+    "emd": {
+        "Home-page": "https://emd.readthedocs.io/en/stable",
+        "Summary": "Empirical Mode Decomposition in Python.",
+    },
+}
+
+REQUIRE_METADATA = os.getenv("MNE_REQUIRE_RELATED_SOFTWARE_INSTALLED", "false").lower()
+REQUIRE_METADATA = REQUIRE_METADATA in ("true", "1")
+
+# These packages pip-install with a different name than the package name
+RENAMES = {
+    "python-neo": "neo",
+    "matplotlib-base": "matplotlib",
+}
+
+_memory = joblib.Memory(location=pathlib.Path(__file__).parent / ".joblib", verbose=0)
+
+
+@_memory.cache(cache_validation_callback=joblib.expires_after(days=7))
+def _get_installer_packages():
+    """Get the MNE-Python installer package list YAML."""
+    with urllib.request.urlopen(
+        # TODO: Set to upstream main once mne-installers reorg PR is merged
+        "https://raw.githubusercontent.com/larsoner/mne-installers/related/recipes/mne-python/construct.yaml"
+    ) as url:
+        data = url.read().decode("utf-8")
+    # Parse data for list of names of packages
+    lines = [line.strip() for line in data.splitlines()]
+    start_idx = lines.index("# <<< BEGIN RELATED SOFTWARE LIST >>>") + 1
+    stop_idx = lines.index("# <<< END RELATED SOFTWARE LIST >>>")
+    packages = [
+        # Lines look like
+        # - mne-ari =0.0.0
+        # or similar.
+        line.split()[1]
+        for line in lines[start_idx:stop_idx]
+        if not line.startswith("#")
+    ]
+    return packages
+
+
+@functools.lru_cache
+def _get_packages():
+    packages = _get_installer_packages()
+    for name in MANUAL_PACKAGES:
+        if name not in packages:
+            packages.append(name)
```

Comment:
```suggestion
    for name in PYPI_PACKAGES | set(MANUAL_PACKAGES):
        if name not in packages:
            packages.append(name)
```

Also this silently allows duplication between installer packages and the list of manual ones. Might be worth warning/erroring to force deduplication?